### ISSUE TITLE
Minor: Standardise variable names

### ISSUE THREAD

**Issue by baolsen** [cattle-ops/terraform-aws-gitlab-runner#575]
Some variable names could be standarised.

First example, all of the below relate to cloudwatch log groups:
`log_group_name`
`enable_cloudwatch_logging`
`create_cloudwatch_log_group`
`cloudwatch_logging_retention_in_days`

Probably best to leave this issue for a little while to collect and discuss other cases of standardisation and then to a single PR to resolve.

It may be helpful to also compile a "migration list" so that users can easily upgrade between different named variables if there are many.

Note, changing these would be breaking changes.

**Comment by kayman-mk** [comment]
First proposal:

```hcl
module "gitlab_runner" {
  agent {
    instance_type = "t3.micro"
    ebs_optimized = true
    enable_monitoring = true

    # et al.
  }

  executor {
    egress_rules = [ ...]
    ami_owners = [ ... ]

    cache_bucket {
    }

    # et al.
  }

  vpc_id = "abc"
  permissions_boundary = "xyz"
  enable_kms = true

  # et al.
}
```

**Comment by kayman-mk** [comment]
I prepared a long lived feature branch for this. See #723

**Comment by kayman-mk** [comment]
@npalm What do you think about the above mentioned restructuring? If you are fine with the `agent` and `executor` blocks I would work on the renaming of the variables.

**Comment by tmeijn** [comment]
Hey @kayman-mk I had a proposal/thought, let me know what you think?

I was actually thinking, why don't remap all variables related to the `config.toml` to Terraform object variables?

We've already been pretty successful imho in #511 and #711 using this pattern, so I was thinking why don't we use the same pattern for all sections of the `config.toml`?

```hcl
module "gitlab_runner" {
# other options....

<good_prefix>_config_global_section = {
  # all options we want configurable in the global section of the `config.toml`. Corresponds to https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-global-section
}

<good_prefix>_config_runners_section = {
  # Corresponds to https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runners-section
}

<good_prefix>_config_runners_docker_section = {
  # Corresponds to https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersdocker-section
}

<good_prefix>_config_runners_machine_section = {
  # Corresponds to https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersmachine-section
}

<good_prefix>_config_runners_machine_section_machine_options = {
  # Corresponds to https://gitlab.com/gitlab-org/ci-cd/docker-machine/-/blob/main/drivers/amazonec2/amazonec2.go#L128
}

https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersdocker-section

}

```

That would:
* reduce a lot of the input variables while making it easier (imo) to add additional properties when to become available
* Make a pretty good mapping between the inputs of this module and the actual GitLab Runner Documentation. We could then more easily just refer the user to the official GitLab documentation.

**Comment by tmeijn** [comment]
I think documenting as "Corresponds to the <section name> in the GitLab Runner config. See <DOC URL> for options and explanation" should be good enough documentation if you do in the proposed way? 🤔 

```hcl
variable "<good_prefix>_config_global_section" {
  description = "Corresponds to the Global section in the GitLab Runner config. See https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-global-section for available options and explanation"
}
```

We could then also add a multi-line doc explaining the case conversion scheme and which options are not available due to whatever reason.

```hcl
variable "<good_prefix>_config_runners_machine_section" {
  description = <<EOF
     Corresponds to the [[runners.machine]] section in the GitLab Runner config. See https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersmachine-section
     
     Note that not all options are available, see the object definition for available options
  EOF
}
```

**Comment by kayman-mk** [comment]
Fixed with the 7.0.0 release of the module.
